staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1364,6 +1365,22 @@ func HasSchemaWith(spec *apiextensions.CustomResourceDefinitionSpec, pred func(s
 	return false
 }
 
+var schemaPool = sync.Pool{
+	New: func() any {
+		return new(apiextensions.JSONSchemaProps)
+	},
+}
```

Comment:
Keeping this global means we keep memory around from the high watermark forever. Would making the pool local to each top-level `SchemaHas` call (would require a tiny bit of refactoring) significantly decrease the huge gains you posted?

Reply:
No, pools release unused objects after (IIRC) 2 GC passes.

Reply:
> The store scales under load (when many goroutines are actively printing) and shrinks when quiescent.

Was hard to find a reference in docs to this and this was closest I could get. After a bit a research it does seem thats the implementation. Thanks for the correction!